### PR TITLE
[17.0][MIG] intrastat_product

### PR DESCRIPTION
**Pull request by alexis-via**
Migration from v16 PR https://github.com/OCA/intrastat-extrastat/pull/260

There are still 6 ruff errors "B023 Function definition does not bind loop variable `rec`", but I'm not sure if they are relevant or not.

### CODE REVIEW

**Review by njeudy — APPROVED** [review]
Test on real database to check if all is working as it does in V16. Don't understand precommit error ...

**Comment by GuillemCForgeFlow** [comment]
Hi @alexis-via could you rebase here so I can review? Module `product_harmonized_system` was not yet added in v17.0

**Comment by alexis-via** [comment]
rebased

File: intrastat_product/models/intrastat_transaction.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    code = fields.Char(required=True)
+    description = fields.Text(translate=True)
+    # intrastat.transaction are shared among companies by default
+    company_id = fields.Many2one("res.company")
+    active = fields.Boolean(default=True)
+
+    @api.depends("code", "description")
+    def _compute_display_name(self):
+        for this in self:
+            name = this.code
```
Comment (GuillemCForgeFlow) [inline intrastat_product/models/intrastat_transaction.py:33]:
```suggestion
            name = this.code or ""
```
Same error as what was fixed here: https://github.com/OCA/intrastat-extrastat/pull/268

File: intrastat_product/models/account_move.py [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+    hs_code_id = fields.Many2one(
+        comodel_name="hs.code",
+        compute="_compute_hs_code_id",
+        string="Intrastat Code",
+    )
+
+    def _compute_hs_code_id(self):
+        for rec in self:
+            intrastat_line = self.move_id.intrastat_line_ids.filtered(
+                lambda r: r.invoice_line_id == rec
```
Comment (GuillemCForgeFlow) [inline intrastat_product/models/account_move.py:152]:
```suggestion
                lambda r, rec=rec: r.invoice_line_id == rec
```
To fix the pre-commit error you need to like so for all cases.

**Review by GuillemCForgeFlow — COMMENTED** [review]
General code review LGTM 👍🏿

**Comment by dreispt** [comment]
Added fixes in #276

**Review by traitrantrobz — APPROVED** [review]
LGTM 👍

**Comment by gurneyalex** [comment]
I triggered a merge of #276

**Pull request merged**